core/server/data/schema/commands.js

```diff
@@ -87,10 +87,21 @@ function createTable(table, transaction) {
             }
 
             return (transaction || db.knex).schema.createTable(table, function (t) {
+                const compositeIndexes = [];
+
                 const columnKeys = _.keys(schema[table]);
                 _.each(columnKeys, function (column) {
+                    const columnSpec = schema[table][column];
+                    if (Object.prototype.hasOwnProperty.call(columnSpec, 'index') && _.isArray(columnSpec.index)) {
+                        compositeIndexes.push(columnSpec.index);
+                    }
+
                     return addTableColumn(table, t, column);
                 });
+
+                _.each(compositeIndexes, function (compositeIndex) {
+                    t.index(compositeIndex);
+                });
```

Comment:
This feels a little hacky but I needed something quick to implement. It allows you to use `index: [...'columns']` in `schema.json` column specs.

One thing I'm not sure about is needing to specify the column that's being defined, eg `mycol: {index: ['mycol', 'othercol']}`. Should the `'mycol'` part of the index be implied as this is being defined in the column definition?

Reply:
It's a tricky one as it doesn't necessarily fit nicely into how keys are defined in sql (as separate definition listing all columns). 

The problem with omitting explicit `'mycol'` in `index` array is that it might imply that this is a partial index definition and the user has to define similar statement on all the columns that take part in composite index definition, e.g.: `othercol: {index: ['mycol']}`. Would personally leave a single index definition listing **all columns** just like in the example above: `mycol: {index: ['mycol', 'othercol']}.`

Reply:
IMO including all columns is preferable too!

Did you consider possibly adding support for something like:
```js
table_name: {
  column_a: {...},
  column_b: {...},
  '@@INDEXES@@': { // Or some other non-conflicting key
    index_name: ['column_a', 'column_b']
  }
}
```

Maybe that's nice as it replicates a bit more how indexes are added in SQL :shrug:

Reply:
love your idea @allouis :heavy_plus_sign: :heavy_plus_sign:

Reply:
`{index_name: ['column_a', 'column_b']}`

We generally rely on knex to create the index names for us. Would `[['column_a', 'column_b'], ['column_a', 'column_c']]` make sense or is that painting ourselves into a corner for when we want to specify our own names?

Reply:
That's true!

I think that array syntax is reasonable, and not really painting ourselves into a corner as we only define these in the schema file, and it wouldn't be much work to change the implementation _if_ we ever want to specify our own names!

Reply:
Updated to use array syntax and `'@@INDEXES@@'` key. Thanks for the input! 🙌